breaking: Support `themed` and `neutral` Tab variations

frankieyan opened the pull request:
## Short description

This changes the `Tab` component's API and drops a few variations that we no longer use. The `color` prop has been removed, and `variant` now accepts `themed` and `neutral` as valid options.

More details are in https://github.com/Doist/Issues/issues/6854 and https://twist.com/a/1585/inbox/t/3688056/.

![image](https://user-images.githubusercontent.com/8531248/184360532-98b8cf00-28a1-41e4-a439-25f10a4c2cf5.png)

## PR Checklist

<!--
Feel free to leave unchecked or remove the lines that are not applicable.
-->

-   [ ] Added tests for bugs / new features
-   [x] Updated docs (storybooks, readme)
-   [x] Executed `npm run validate` and made sure no errors / warnings were shown
-   [x] Described changes in `CHANGELOG.md`
-   [ ] Bumped version in `package.json` and `package-lock.json` (`npm --no-git-tag-version version <major|minor|patch>`) [ref](https://docs.npmjs.com/cli/v6/commands/npm-version)
-   [ ] Updated all static build artifacts (`npm run build-all`)

## Versioning

Major

rfgamaral reviewed (approved):
Looks good to me, although I'm curious about the introduction of `Required`. Doesn't this introduce a breaking change? Do we want that?

frankieyan commented:
> Looks good to me, although I'm curious about the introduction of `Required`. Doesn't this introduce a breaking change? Do we want that?

<img width="366" alt="image" src="https://user-images.githubusercontent.com/8531248/184317523-2956f9ca-4c89-4191-bbcf-b56a075cf890.png">

@rfgamaral you mean this change right? That is for the context internal to the component, and not the component's props themselves which remain optional. Since we have default values for those props they should never end up being undefined in the context. The only thing it does is it saves us from doing a `contextValue.space ?? ''` when trying to use them as strings to get the css classnames.

rfgamaral commented:
Yes, that's what I was referring to. Thanks for the clarification.

frankieyan commented:
Hi @rfgamaral, sorry I've updated the component after a discussion with the design team [here](https://twist.com/a/1585/inbox/t/3688056/), and this is now a breaking change as I've removed older variants that we will no longer use. Can you please take another look :)

rfgamaral reviewed (approved):
Looks good to me 🚀
